### PR TITLE
ICU-20979 BRS 67rc, ICU4C API promotion

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20979
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

ICU 67 C API promotion per spreadsheet in https://docs.google.com/spreadsheets/d/1EryaBvIZkiGRJLZe37PHlCv_Lz2MPg0WEaNnV7pUTiw

Also fixes use of U_INTERNAL and U_DRAFT in a couple of files per comments in https://github.com/unicode-org/icu/pull/1072

### CODE REVIEW

File: icu4c/source/common/ustr_imp.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ ustr_hashICharsN(const char *str, int32_t length);
  * @return If UChar is a lowercase ASCII character, returns the uppercase version.
  *         Otherwise, returns the input character.
  */
-U_CAPI UChar U_EXPORT2
+U_INTERNAL UChar U_EXPORT2
```
Comment (jefgen) [inline icu4c/source/common/ustr_imp.h:56]: Thank you for fixing this and also for changing the others in the file as well!

File: icu4c/source/i18n/unicode/uformattedvalue.h [hunk 2]
```diff
@@ -120,7 +118,7 @@ typedef struct UConstrainedFieldPosition UConstrainedFieldPosition;
  *
  * @param ec Set if an error occurs.
  * @return The new object, or NULL if an error occurs.
- * @draft ICU 64
+ * @stable ICU 64
  */
 U_DRAFT UConstrainedFieldPosition* U_EXPORT2
```
Comment (jefgen) [inline icu4c/source/i18n/unicode/uformattedvalue.h:123]: nit: As you noted on the other PR: U_DRAFT = U_CAPI. :)
Though perhaps U_STABLE for documentation purposes?

Comment (pedberg-icu) [reply icu4c/source/i18n/unicode/uformattedvalue.h:123]: Oh, good catch, thanks, I am fixing several instances of this now in uformattedvalue.h, unumberformatter.h, ureldatefmt.h

**Comment by pedberg-icu** [comment]
The CI headertest shows an error, I need to fix this:
```
drafttest unicode/localebuilder.h
12858In file included from ht_stub_drafttest.cpp:1:
12859/tmp/icu-prefix/include/unicode/localebuilder.h:295:18: error: ‘LocaleMatcher’ has not been declared
12860     friend class LocaleMatcher::Result;
12861                  ^~~~~~~~~~~~~
12862/tmp/icu-prefix/include/unicode/localebuilder.h:295:5: error: friend declaration does not name a class or function
12863     friend class LocaleMatcher::Result;
12864     ^~~~~~
```

**Comment by markusicu** [comment]
I restarted the Travis Mac OS build.

**Comment by pedberg-icu** [comment]
OK, squashed. Headertest passes. On config test had some kind of config/start problem (error generating build script, seems unrelated). Need re-approval to merge, thanks.

**Comment by jefgen** [comment]
FWIW, the pipelines MacOSX build did succeed:
https://unicode-icu.visualstudio.com/ICU/_build/results?buildId=7680&view=logs&j=ae7cbf3c-29a1-57e4-6058-0dde69620123
(Though it only builds, it doesn't run tests).
I'm not sure what's wrong with the Travis build. it just says "An error occurred while generating the build script." -- which isn't really helpful...

**Pull request merged**